crates/db_schema/src/aggregates/community_aggregates.rs

```diff
@@ -167,8 +167,8 @@ mod tests {
     // Remove a parent post (the comment count should also be 0)
     Post::delete(pool, inserted_post.id).await?;
     let after_parent_post_delete = CommunityAggregates::read(pool, inserted_community.id).await?;
-    assert_eq!(0, after_parent_post_delete.comments);
     assert_eq!(0, after_parent_post_delete.posts);
+    assert_eq!(0, after_parent_post_delete.comments);
```

Comment:
Theres a test failure on this line but I cant figure out why. The relevant trigger seems to be in [r.update_comment_count_from_post ()](https://github.com/LemmyNet/lemmy/pull/5407/files#diff-4bf296e1c1010278693efa6351d96dbd2678b912c0095e305c1cd2bf3d4566e7R311), but there are no changes except for simple renaming.

@dessalines @dullbananas

Reply:
At a first glance, that function is doing an inner join from post to post now, where it should only need to do a `where` filter. 

@dullbananas when you get a chance, could you go over this PR and make any corrections you see necessary to the replaceable schema utils and triggers?

Reply:
Hmm I tried to change that but still not working. And yes triggers definitely need a close review before merging.

Reply:
Anyway I will keep working to remove the remaining aggregate tables (and likely break more stuff). You can use [this commit](https://github.com/LemmyNet/lemmy/pull/5407/commits/f2d28523b9051e5cd546de5d680f12b816dee900) to look into this test failure.

Reply:
The comment trigger had infinite recursion because it updated parent comments, and statetement-level triggers always run even if zero rows are updated. The `pg_trigger_depth( ) = 0` condition prevents the comment trigger from running if the comment was deleted by the `delete_comments_before_post` trigger. Remove the depth condition and check for `(SELECT count(*) FROM select_old_and_new_rows AS old_and_new_rows) = 0` in the trigger function so it can return early. This is less error prone than increasing the depth limit.

Reply:
Its not clear to me, where exactly should I put the check `(SELECT count(*) FROM select_old_and_new_rows AS old_and_new_rows) = 0;`?

Reply:
The condition needs to be checked in the function body using an if statement that contains a return statement

Reply:
I'm trying to fix the triggers right now